logs_subscriber.py

```diff
@@ -0,0 +1,79 @@
+# subscribe the Axiom ingester to newly created log groups
+import boto3
+import os
+import logging
+
+# Set environment variables.
+axiom_cloudwatch_lambda_ingester_arn = os.getenv("AXIOM_CLOUDWATCH_LAMBDA_INGESTER_ARN")
+log_group_prefix = os.getenv("LOG_GROUP_PREFIX", "")
+
+# set logger
+level = os.getenv("log_level", "INFO")
+logging.basicConfig(level=level)
+logger = logging.getLogger()
+logger.setLevel(level)
+
+# Set up CloudWatch Logs client.
+log_client = boto3.client("logs")
+lambda_client = boto3.client("lambda")
+
+
+def lambda_handler(event, context):
+    """
+    Subscribes log ingester to log group from event.
+
+    :param event: Event data from CloudWatch Logs.
+    :type event: dict
+
+    :param context: Lambda context object.
+    :type context: obj
+
+    :return: None
+    """
+    if not "detail" in event:
+        return
+    # Grab the log group name from incoming event.
+    aws_account_id = event["account"]
+    aws_region = event["detail"]["awsRegion"]
+    log_group_name = event["detail"]["requestParameters"]["logGroupName"]
+    log_group_arn = (
+        f"arn:aws:logs:{aws_region}:{aws_account_id}:log-group:{log_group_name}:*"
+    )
+
+    # Check whether the prefix is set - the prefix is used to determine which logs we want.
+    if not log_group_prefix:
+        create_subscription_filter(
+            log_group_name, log_group_arn, axiom_cloudwatch_lambda_ingester_arn
+        )
+
+        # Check whether the log group's name starts with the set prefix.
+    elif log_group_name.startswith(log_group_prefix):
```

Comment:
no difference between if log_group_prefix set or not set so you could just put in one condition.
also please add  else -> log that it didn't match the log prefix

Reply:
done